webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchService.java

```diff
@@ -170,14 +170,21 @@ private void archiveFile(final MultipartFile file,
             }
 
             validationFailures.addAll(validationService.validateHeaders(parser.getHeaderMap().keySet()));
-
             validationFailures.addAll(validationService.validateRecords(recordIterator, permissionScope));
 
         } catch (final IOException ex) {
+            logger.warn("Problem reading uploaded file", ex);
             validationFailures.add(CsvValidationFailure.builder()
                     .row(0)
                     .message("Error reading file [" + ex.getMessage() + "]")
                     .build());
+        } catch (final IllegalArgumentException ex) {
+            logger.warn("Problem parsing uploaded file", ex);
+            validationFailures.clear();
+            validationFailures.add(CsvValidationFailure.builder()
+                    .row(0)
+                    .message("File does not appear to be a valid CSV with a header row")
+                    .build());
```

Comment:
When the CSVParser encounters bad data (say, from a .docx or .mp4 file) it throws an IllegalArgumentException containing data from the source file.  To protect against having a lot of unknown characters in the response message, I'm going to log the error and return a generic "probably not a CSV with a header row" response message.